ap/production/normalization.py

```diff
@@ -0,0 +1,72 @@
+# coding: utf-8
+
+"""
+Column production methods related to sample normalization event weights.
+"""
+
+import law
+import order as od
+
+from ap.production import Producer, producer
+from ap.util import maybe_import
+from ap.columnar_util import set_ak_column
+
+ak = maybe_import("awkward")
+
+
+@producer(
+    uses={"LHEWeight.originalXWGTUP"},
+    produces={"normalization_weight"},
+)
+def normalization_weights(
+    events: ak.Array,
+    selection_stats: dict,
+    config_inst: od.Config,
+    dataset_inst: od.Dataset,
+    **kwargs,
+) -> ak.Array:
+    """
+    Uses luminosity information of *config_inst*, the cross section of a process obtained through
+    *dataset_inst* and the sum of event weights from *selection_stats* to assign each event a
+    normalization weight.
+    """
+    # get lumi and cross section of the first process contained in the dataset
+    # TODO: the process should depend on the process id per event for which we need a LUT
+    #       that is cached through a simple mutable argument
+    lumi = config_inst.x.luminosity.nominal
+    xs = dataset_inst.processes.get_first().get_xsec(config_inst.campaign.ecm).nominal
+
+    # compute the weight and store it
+    norm_weight = events.LHEWeight.originalXWGTUP * lumi * xs / selection_stats["sum_mc_weight"]
```

Comment:
I might need to check that again, but I think the process cross section is already taken into account by the LHEWeight.

Reply:
Oh nevermind, you are dividing by the sum of weights, so this is already taken into account

Reply:
Oh I hope not. The cross sections from the database are usually not the ones one wants to use.

I remember from (mini)AOD times that "originalXWGTUP" used to be the generator dependent one without xs. I hope this didn't change with nanoAOD ...